### PR TITLE
Added rough MacOS backend

### PR DESCRIPTION
**Pull request by SamoZ256**
This commit adds a rough MacOS backend. It only works with Vulkan. A lot of it still does not work, but the vkcube example runs fine. I had to change some build files, which may break it for Linux (namely adding objc as a language in the main build file). This is my first time working with Meson, so sorry for that. Also, there's a hardcoded path to the libwsi-cocoa.dylib file, but I will change that soon.

### CODE REVIEW

**Comment by markbolhuis** [comment]
Thanks very much!

Few things:
- The public API of this library is still undecided, and is going to change dramatically so please don't add anything to it.
However having multiple backends will help decide on what works cross-platform. 
- The wrapper library isn't necessary on macOS. It's only needed on Linux to switch between x11 to Wayland. Instead you can just build a single library: `libwsi.dynlib`. I might move the x11 and wayland directories, and dispatcher library files into a linux directory to reflect this. 
- I'm debating if i should stick with meson, and instead use CMake, so don't worry about that. I want to have a win32 backend as well and CMake is much more friendly on windows.
- How to handle OpenGL context creation is undecided. I'm thinking of just exposing a way of creating the native context objects, and not try and wrap them.

**Comment by SamoZ256** [comment]
Thank you for your reply.

So if I understand correctly, you don’t accept pull requests right now? Regarding CMake, I think that’s a good idea, since many more people are familiar with it.

**Comment by markbolhuis** [comment]
> So if I understand correctly, you don’t accept pull requests right now?

Not until I've made some changes to the repo. I need to make it more friendly to other backends. I see you've have to make edits in the meson.build files. However I'd like those changes to be in a separate commit. I'll push one to update them myself later.

I'd also like to discuss in depth what features, requirements and limitations cocoa has before just accepting a PR. I am completely unfamiliar with it, so feel free to open an issue on this.

File: meson.build [hunk 1]
```diff
@@ -1,4 +1,4 @@
-project('libwsi', 'c',
+project('libwsi', 'c', 'objc',
```
Comment (markbolhuis) [inline meson.build:1]: Can use `add_languages('objc')`

https://mesonbuild.com/Reference-manual_functions.html#add_languages

**Comment by SamoZ256** [comment]
Ok so should I update it to work with your newest changes?

**Pull request closed without merging**